### PR TITLE
write tests that assert the markers are shown or hidden as in the new example code

### LINKED ISSUE #45: write tests that assert the markers are shown or hidden as in the new example code!
Hi @jywarren ,

Can you give some examples from other repositories ? (similar examples would also work 😄 !)

Thank you 🎈 !

### PR DESCRIPTION
**Pull request by sagarpreet-chadha**
Fixes #45 !

### CODE REVIEW

**Comment by sagarpreet-chadha** [comment]
Hi @jywarren !

`dist/Leaflet.BlurredLocationDisplay.js` is not being loaded before the tests are loaded . Can you help in setting  up the testing environment ? Thanks !

screenshot 2019-03-08 at 5 03 21 pm

**Comment by sagarpreet-chadha** [comment]
Hi @jywarren , 
Kindly take a look at it once 😄 !

**Comment by sagarpreet-chadha** [comment]
Yes tried that as well ! 
Added LBL , LBLD dependencies in vendor , still showing the same error .
LBLD is not getting loaded before the specs starts running .

**Comment by jywarren** [comment]
Hmm, confusing. Maybe you have to include the source files like this?

https://github.com/publiclab/PublicLab.Editor/blob/master/Gruntfile.js#L44-L46

If this doesn't work, maybe ping `@rexagod` or someone else from `image-sequencer` as they are really good with test setup!

**Comment by sagarpreet-chadha** [comment]
Hey @rexagod , @tech4GT , @publiclab/reviewers ! 
Can someone with good testing background help me solve this one ? Looking forward to hear from you guys :smile: . Thank you :)

**Comment by rexagod** [comment]
Noticing that Travis and local both give different errors (`BlurredLocation` and `BlurredLocationDisplay` respectively). @sagarpreet-chadha Are you getting an `asynchronous` warning in your logs as well?

Working on this.

File: spec/javascripts/test_spec.js [hunk 1]
```diff
@@ -1,5 +1,31 @@
...
+
+ var fixture = loadFixtures('example.html');
+
   it("Basic Test", function () {
   expect(true).toBe(true);
   });
+
+  it("Checks if at zoom level 5 , all 7 markers are shown", function () {
+    //BlurredLocation1.map.setZoom(5);
+    expect($("#map1").children()[0].childNodes[3].childNodes.length).toBe(0) ;
```
Comment (rexagod) [inline spec/javascripts/test_spec.js]: ` expect($("#map1").children()[0].childNodes[3].childNodes.length).toBe(7) ;`

File: spec/javascripts/test_spec.js [hunk 2]
```diff
@@ -1,5 +1,31 @@
...
+  });
+
+  it("Checks if at zoom level 6 , only 5 markers are shown", function () {
+    //BlurredLocation2.map.setZoom(6);
+    expect($("#map2").children()[0].childNodes[3].childNodes.length).toBe(0) ;
```
Comment (rexagod) [inline spec/javascripts/test_spec.js]: ` expect($("#map1").children()[0].childNodes[3].childNodes.length).toBe(5) ;`

File: spec/javascripts/test_spec.js [hunk 3]
```diff
@@ -1,5 +1,31 @@
...
+  });
+
+  it("Checks if at zoom level 8 , only 3 markers are shown", function () {
+   // BlurredLocation3.map.setZoom(8);
+    expect($("#map3").children()[0].childNodes[3].childNodes.length).toBe(0) ;
```
Comment (rexagod) [inline spec/javascripts/test_spec.js]: ` expect($("#map1").children()[0].childNodes[3].childNodes.length).toBe(3) ;`

**Comment by rexagod** [comment]
Hello, @sagarpreet-chadha! I've made some comments above, depending on some checks I made below.

Screenshot from 2019-03-22 23-51-19

**Comment by rexagod** [comment]
Getting contradicting results.

Screenshot from 2019-03-23 17-43-01

Screenshot from 2019-03-23 17-42-47

Looking into this.

**Comment by rexagod** [comment]
Made some changes in the way.

https://github.com/publiclab/leaflet-blurred-location-display/blob/246b4da88847ccb7f8fc6e29d943911bfacd8dc1/src/ui/gridCenterRectangle.js#L43

-> `if(map && map.getZoom() >= 3 && map.getZoom() <=9)`, since getZoom isn't initialized `onDOMload` so we might want to look into that as well.

https://github.com/publiclab/leaflet-blurred-location-display/blob/246b4da88847ccb7f8fc6e29d943911bfacd8dc1/src/ui/gridCenterRectangle.js#L35

-> `if(typeof options.blurredLocation.getRectangle !== "undefined")`, removed parenthesis here, since `getRectangle` itself wasn't on `blurredLocation`'s `__proto__`.

**Comment by rexagod** [comment]
Looking into this.

File: spec/javascripts/test_spec.js (hunk 1 above)
Comment (sagarpreet-chadha) [reply spec/javascripts/test_spec.js]: Hi @rexagod , yes i did that purposely to tell you guys that in testing , markers are not getting loaded on map 😄 !

**Comment by sagarpreet-chadha** [comment]
Hi @rexagod ...thank you for the recommendations 😄 . What do you think could be the possible reason that LBL and LBLD libraries are not getting loaded before testing starts ? Thanks !

**Comment by sagarpreet-chadha** [comment]
Thank you @arpansac , the fixtures are now being loaded 🎉 .
2 out of 5 tests are running now 👍 .

Also now all libraries are also getting loaded except LBLD .
Screenshot 2019-03-28 at 10 52 33 PM

Now we just have to figure why LBLD is not getting loaded . Thanks all 😄 !

**Comment by sagarpreet-chadha** [comment]
We use Jasmine jquery to load fixtures , but this package is not maintained . Also see this which i found in its documentation (this should solve the error which one gets when `grunt jasmine` is run on localhost) : 
Source : https://github.com/velesin/jasmine-jquery
Screenshot 2019-04-01 at 3 32 08 PM

And this error on Travis 😄 asks to open issue in puppeteer package  : 

Screenshot 2019-04-01 at 3 43 04 PM

**Comment by sagarpreet-chadha** [comment]
grunt-contrib-Jasmine runs on phantomJS , and phantomJS does not support ES6 !

**Comment by sagarpreet-chadha** [comment]
YAYYY 🎉 

I think i will write a blog about it :P .
So some ES6 functions were causing problem because phantomJS does not support ES6 !
And regarding LBLD not loading in fixtures --- it is yet an unsolved issue in Jasmine-jquery !

@jywarren ...kindly review 😄 !  
Should we add put this library to npm ?

**Comment by sagarpreet-chadha** [comment]
@jywarren ...kindly review this one as well 😄 !

**Comment by jywarren** [comment]
OK! Sorry i had replied via email but it didn't go through!

**Comment by jywarren** [comment]
Awesome! Thanks Sagarpreet and great problem solving, this was such a tough one. I'm sorry i wasn't able to help more! Perhaps a good deal of this testing workflow is starting to age a bit. But you got it! 🙌

**Pull request merged**